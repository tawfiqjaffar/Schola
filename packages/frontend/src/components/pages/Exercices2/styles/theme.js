import gray from "@material-ui/core/colors/grey";
import lightGreen from "@material-ui/core/colors/lightGreen";

import { createMuiTheme } from "@material-ui/core/styles";

const theme = createMuiTheme({
    palette: {
        primary: {
            light: lightGreen[200],
            main: lightGreen[400],
            dark: lightGreen[600],
            contrastText: "#fff"
        },
        secondary: {
            light: gray[200],
            main: gray[500],
            dark: gray[700],
            contrastText: "#fff"
        }
    }
});

export default theme;
