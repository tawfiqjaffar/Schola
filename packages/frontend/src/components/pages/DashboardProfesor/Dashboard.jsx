import React from 'react';
import clsx from 'clsx';
import { Container, Grid, Paper } from '@material-ui/core';
import { makeStyles } from '@material-ui/styles';
import BlueBackground from '../../common/BlueBackground';

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    textAlign: 'center',
  },
  paper: {
    marginTop: theme.spacing(8),
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
    paddingLeft: theme.spacing(3),
    paddingRight: theme.spacing(3),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    background: 'white'
  },
  fixedHeight: {
    height: '240px',
  },
  paperphoto: {
    height: '270px',
    width: '200px'
  },
  paperinfos: {
    height: '270px',
    width: '200px'
  },
  papernotifs: {
    height: '310px',
    width: '420px'
  },
  papercalendar: {
    height: '340px',
    width: '600px'
  },
  papergrades: {
    height: '330px',
    width: '330px'
  },
  background: {
    background: 'white',
  },
}));

const Dashboard = () => {
  const classes = useStyles();

  const IdPhoto = clsx(
    classes.paperphoto,
    classes.background
  );

  const Personal = clsx(
    classes.paperinfos,
    classes.background
  );

  const Notifications = clsx(
    classes.papernotifs,
    classes.background
  );

  const Calendar = clsx(
    classes.papercalendar,
    classes.background
  );

  const Moyennes = clsx(
    classes.papergrades,
    classes.background
  );

  return (
    <Container className={classes.root}>
      <Paper className={classes.paper}>
        <Grid container spacing={3}>

          <BlueBackground Bheight="315" Bwidth="260" Fheight="315" Fwidth="260">
            <Paper className={IdPhoto}>
              <h3>Mes infos</h3>
              <h4>Nom prénom</h4>
              <h4>Classe:</h4>
              <h4>Nombre d'élèves:</h4>
              <h4>Élève responsable de la classe:</h4>
            </Paper>
          </BlueBackground>

          <BlueBackground Bheight="315" Bwidth="260" Fheight="315" Fwidth="260" style={{ marginLeft: '80px' }}>
            <Paper className={Personal}>
              <h3>Bloc-note</h3>
            </Paper>
          </BlueBackground>

          <BlueBackground Bheight="315" Bwidth="490" Fheight="315" Fwidth="490" style={{ marginLeft: '80px' }}>
            <Paper className={Notifications}>
              <h3>Messagerie</h3>
            </Paper>
          </BlueBackground>

          <BlueBackground Bheight="340" Bwidth="690" Fheight="340" Fwidth="680" style={{ marginTop: '60px' }}>
            <Paper className={Calendar}>
              <h3>Création évenments</h3>
            </Paper>
          </BlueBackground>

          <BlueBackground Bheight="350" Bwidth="400" Fheight="350" Fwidth="400" style={{ marginLeft: '100px', marginTop: '40px' }}>
            <Paper className={Moyennes}>
              <h3>Création exercices</h3>
            </Paper>
          </BlueBackground>

        </Grid>
      </Paper>
    </Container>
  );
};
export default Dashboard;
