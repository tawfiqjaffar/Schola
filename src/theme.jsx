import { createMuiTheme } from '@material-ui/core/styles';

const palette = {
  primary: { main: '#70C6C7', contrastText: '#212121' },
  secondary: { main: '#60c07a', contrastText: '#FAFAFA' },
  text: {
    primary: '#212121',
    secondary: '#FAFAFA',
  },
  background: {
    paper: '#70C6C7',
  },
};
const themeName = 'Downy Pastel Green Gayal';

export default createMuiTheme({ palette, themeName });
