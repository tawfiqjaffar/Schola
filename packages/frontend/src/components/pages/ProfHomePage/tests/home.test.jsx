import React from 'react';

import renderer from 'react-test-renderer';

import { ThemeProvider } from '@material-ui/core';
import { BrowserRouter } from 'react-router-dom';
import Dashboard from '../Dashboard';
import theme from '../../../../theme';

describe('Dashboard', () => {
  it('should render Profesor Dashboard', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <Dashboard />
        </BrowserRouter>
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});