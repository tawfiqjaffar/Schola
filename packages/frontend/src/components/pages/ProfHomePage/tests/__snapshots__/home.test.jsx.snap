// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render Profesor Dashboard 1`] = `
<div
  className="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        className=""
        style={
          Object {
            "backgroundColor": "#9ed1cc",
            "height": "315px",
            "transform": "rotate(3deg)",
            "width": "260px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#70c6c7",
              "display": "flex",
              "height": "315px",
              "justifyContent": "center",
              "left": "50%",
              "position": "absolute",
              "top": "50%",
              "transform": "translate(-50%, -50%) rotate(-10deg)",
              "width": "260px",
            }
          }
        >
          <div
            style={
              Object {
                "transform": "rotate(7deg)",
              }
            }
          >
            <div
              className="MuiPaper-root makeStyles-paperphoto-4 makeStyles-background-9 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h3>
                Mes infos
              </h3>
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
        style={
          Object {
            "backgroundColor": "#9ed1cc",
            "height": "315px",
            "marginLeft": "80px",
            "transform": "rotate(3deg)",
            "width": "260px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#70c6c7",
              "display": "flex",
              "height": "315px",
              "justifyContent": "center",
              "left": "50%",
              "position": "absolute",
              "top": "50%",
              "transform": "translate(-50%, -50%) rotate(-10deg)",
              "width": "260px",
            }
          }
        >
          <div
            style={
              Object {
                "transform": "rotate(7deg)",
              }
            }
          >
            <div
              className="MuiPaper-root makeStyles-paperinfos-5 makeStyles-background-9 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h3>
                Bloc-note
              </h3>
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
        style={
          Object {
            "backgroundColor": "#9ed1cc",
            "height": "315px",
            "marginLeft": "80px",
            "transform": "rotate(3deg)",
            "width": "490px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#70c6c7",
              "display": "flex",
              "height": "315px",
              "justifyContent": "center",
              "left": "50%",
              "position": "absolute",
              "top": "50%",
              "transform": "translate(-50%, -50%) rotate(-10deg)",
              "width": "490px",
            }
          }
        >
          <div
            style={
              Object {
                "transform": "rotate(7deg)",
              }
            }
          >
            <div
              className="MuiPaper-root makeStyles-papernotifs-6 makeStyles-background-9 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h3>
                Messagerie
              </h3>
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
        style={
          Object {
            "backgroundColor": "#9ed1cc",
            "height": "340px",
            "marginTop": "60px",
            "transform": "rotate(3deg)",
            "width": "690px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#70c6c7",
              "display": "flex",
              "height": "340px",
              "justifyContent": "center",
              "left": "50%",
              "position": "absolute",
              "top": "50%",
              "transform": "translate(-50%, -50%) rotate(-10deg)",
              "width": "680px",
            }
          }
        >
          <div
            style={
              Object {
                "transform": "rotate(7deg)",
              }
            }
          >
            <div
              className="MuiPaper-root makeStyles-papercalendar-7 makeStyles-background-9 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h3>
                Création évenments
              </h3>
            </div>
          </div>
        </div>
      </div>
      <div
        className=""
        style={
          Object {
            "backgroundColor": "#9ed1cc",
            "height": "350px",
            "marginLeft": "100px",
            "marginTop": "40px",
            "transform": "rotate(3deg)",
            "width": "400px",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#70c6c7",
              "display": "flex",
              "height": "350px",
              "justifyContent": "center",
              "left": "50%",
              "position": "absolute",
              "top": "50%",
              "transform": "translate(-50%, -50%) rotate(-10deg)",
              "width": "400px",
            }
          }
        >
          <div
            style={
              Object {
                "transform": "rotate(7deg)",
              }
            }
          >
            <div
              className="MuiPaper-root makeStyles-papergrades-8 makeStyles-background-9 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <h3>
                Création exercices
              </h3>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
